packages/flutter_tools/test/commands.shard/hermetic/analyze_continuously_test.dart

```diff
@@ -228,14 +233,73 @@ void main() {
           ],
           completer: completer,
           stdin: IOSink(stdin.sink),
+          stdout: '''
+{"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
+{"event":"analysis.errors","params":{"file":"/directoryA/foo","errors":[{"type":"TestError","message":"It's an error.","severity":"warning","code":"500","location":{"file":"/directoryA/foo","startLine": 100,"startColumn":5,"offset":0}}]}}
+{"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
+'''
         ),
       ]);
 
     final Artifacts artifacts = Artifacts.test();
     final AnalyzeCommand command = AnalyzeCommand(
       terminal: Terminal.test(),
       artifacts: artifacts,
-      logger: BufferLogger.test(),
+      logger: logger,
+      platform: FakePlatform(),
+      fileSystem: fileSystem,
+      processManager: processManager,
+    );
+
+    final TestFlutterCommandRunner commandRunner = TestFlutterCommandRunner();
+    commandRunner.addCommand(command);
+    unawaited(commandRunner.run(<String>['analyze', '--watch']));
+
+    Timer.periodic(const Duration(milliseconds: 10), (Timer timer) {
+      if (logger.statusText.contains('analyzed 1 file')) {
```

Comment:
I couldn't think of a better way to wait for the `AnalysisServer` to finish processing the errors other than spinning and checking stdout for the completion log.  Open to suggestions.

Reply:
Can you use `FakeAsync` and check after each pump of the message loop.

Reply:
Oh that's way better.